### PR TITLE
JCA Arms Dealer

### PR DESCRIPTION
**Pull request by Azodion**
## What type of PR is this?
1. [ ] Bug
2. [x] Change
3. [x] Enhancement
4. [ ] Miscellaneous

### What have you changed and why?
Information:
Added JCA Infantry Arsenal to Arms Dealer - [Main Mod](https://steamcommunity.com/sharedfiles/filedetails/?id=3333302397)
### Please specify which Issue this PR Resolves (If Applicable).
"This PR closes #XXXX!"

### Please verify the following.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?

1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps:

********************************************************
Notes:
Works

### CODE REVIEW

File: A3A/addons/ultimate/config/trader/cfgTraderWeapons.hpp [hunk 1]
```diff
@@ -141,4 +141,8 @@
         class weapons_cwriii : weapons_base
         {
             prefix = "cwrstock";
+        };
+        class JCAArsenal_Compat : weapons_base
```
Comment (Silence-112) [inline A3A/addons/ultimate/config/trader/cfgTraderWeapons.hpp]:
```suggestion
        class weapons_jca : weapons_base
```

File: A3A/addons/ultimate/config/trader/cfgTraderAddons.hpp [hunk 2]
```diff
@@ -261,4 +261,9 @@
     {
         addons[] = {"sab_navallegends"};
         vehicles = "vehicles_navleg";
+    };
+    class addons_JCAArsenal : addons_base
+    {
+        addons[] = {"Weapons_F_JCA_IA","Weapons_F_JCA_IA_Accessories","Weapons_F_JCA_IA_LongRangeRifles_AWM","Weapons_F_JCA_IA_Pistols_P226","Weapons_F_JCA_IA_Pistols_P320","Weapons_F_JCA_IA_Rifles_M4A1","Weapons_F_JCA_IA_Rifles_M4A4","Weapons_F_JCA_IA_Rifles_SR10","Weapons_F_JCA_IA_Rifles_SR25","Weapons_F_JCA_IA_SMGs_MP5"};
+        weapons = "JCAArsenal_Compat";
```
Comment (Silence-112) [inline A3A/addons/ultimate/config/trader/cfgTraderAddons.hpp]:
```suggestion
        weapons = "weapons_jca";
```

File: A3A/addons/ultimate/config/trader/cfgTraderAddons.hpp [hunk 3]
```diff
@@ -261,4 +261,9 @@
     {
         addons[] = {"sab_navallegends"};
         vehicles = "vehicles_navleg";
+    };
+    class addons_JCAArsenal : addons_base
```
Comment (Silence-112) [inline A3A/addons/ultimate/config/trader/cfgTraderAddons.hpp]:
```suggestion
    class addons_jca : addons_base
```

File: A3A/addons/hals/Addons/store/config.hpp [hunk 4]
```diff
@@ -830,5 +831,20 @@ class cfgHALsStore
 				"opticsCWR"
 			};
 		};
+		class JCA_Arsenal_Stock
```
Comment (Silence-112) [inline A3A/addons/hals/Addons/store/config.hpp]:
```suggestion
		class jca_stock
```

File: A3A/addons/ultimate/config/trader/cfgTraderWeapons.hpp [hunk 5]
```diff
@@ -141,4 +141,8 @@
...
+        {
+            prefix = "JCA_Arsenal_Stock"
```
Comment (Silence-112) [inline A3A/addons/ultimate/config/trader/cfgTraderWeapons.hpp]:
```suggestion
            prefix = "jca_stock"
```

**Review by Silence-112 — CHANGES_REQUESTED** [review]
Just needs naming consistency, is otherwise fine

**Comment by wersal454** [comment]
I believe most(if not all) of this stuff is in aegis, but I think it won't hurt if it will double

**Review by Silence-112 — APPROVED** [review]
lgtm

**Pull request merged**